purchase_deposit/models/account_move.py

```diff
@@ -0,0 +1,16 @@
+# Copyright 2023 Quartile Limited (https://www.quartile.co)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    def action_post(self):
+        res = super().action_post()
+        deposit_lines = self.line_ids.filtered(lambda x: x.purchase_line_id.is_deposit)
+        for line in deposit_lines:
```

Comment:
why run two for when you can do it in one for?
```suggestion
        for line in self.line_ids:
            if not line.purchase_line_id.is_deposit:
                continue
      
```

Reply:
I think adding one extra line for deposit_lines is better than iteration for unnecessary lines and adding condition inside loop.

Reply:
Sorry, you right. I lacked this knowledge https://github.com/OCA/purchase-workflow/pull/904#discussion_r419240295.